src/main/java/com/example/petbuddybackend/service/offer/OfferService.java

```diff
@@ -249,4 +264,84 @@ private void editConfigurationSelectedOptions(OfferConfiguration editingConfigur
         }
     }
 
+    @Transactional
+    public List<OfferDTO> setAvailabilityForOffers(CreateOffersAvailabilityDTO createOffersAvailability,
+                                                   String caretakerEmail) {
+
+        assertAvailabilityRangesNotOverlapping(createOffersAvailability.availabilityRanges());
+        List<Offer> modifiedOffers = createOffersAvailability.offerIds()
+                .stream()
+                .map(offerId -> setAvailabilityForOffer(offerId, createOffersAvailability.availabilityRanges(), caretakerEmail))
+                .toList();
+
+        return offerRepository.saveAll(modifiedOffers)
+                .stream()
+                .map(offerMapper::mapToOfferDTO)
+                .toList();
+    }
+
+    private Offer setAvailabilityForOffer(Long offerId, List<AvailabilityRangeDTO> availabilityRanges, String caretakerEmail) {
+
+        Offer offerToModify = getOffer(offerId);
+        assertOfferIsModifyingByOwnerCaretaker(offerToModify, caretakerEmail);
+        Set<Availability> availabilities = createAvailabilities(availabilityRanges, offerToModify);
+
+        if(offerToModify.getAvailabilities() == null) {
+            offerToModify.setAvailabilities(new HashSet<>());
+        }
+        removeAllAvailabilities(offerToModify.getAvailabilities());
+        offerToModify.getAvailabilities().addAll(availabilities);
+
+        return offerToModify;
+    }
+
+    private void removeAllAvailabilities(Set<Availability> availabilities) {
+
+        Set<Availability> copyOfAvailabilities = new HashSet<>(availabilities);
+
+        for(Availability availability : copyOfAvailabilities) {
+            availabilities.remove(availability);
+        }
+
+    }
```

Comment:
```suggestion
    private Offer setAvailabilityForOffer(Long offerId, List<AvailabilityRangeDTO> availabilityRanges, String caretakerEmail) {

        Offer offerToModify = getOffer(offerId);
        assertOfferIsModifyingByOwnerCaretaker(offerToModify, caretakerEmail);
        Set<Availability> availabilities = createAvailabilities(availabilityRanges, offerToModify);

        if(offerToModify.getAvailabilities() == null) {
            offerToModify.setAvailabilities(new HashSet<>());
        }
        offerToModify.setAvailabilities(availabilities);

        return offerToModify;
    }
```

`offerToModify.setAvailabilities(availabilities);` nie da tego samego efektu? Bo właściwie usuwasz wszyskie elementy i dodajesz wszyskie do listy

Reply:
Plus, jak wyżej, możnaby new HashSet przenieść do Entity

```java
if(offerToModify.getAvailabilities() == null) {
    offerToModify.setAvailabilities(new HashSet<>());
}
```

Reply:
no wlasnie przez to sie pierdolilem bo sie wywalal blad z orphanRemoval i wyczytalem ze trzeba recznie wszystko remove i addAll. Nie moge przestawic referencji na nowy set ;/

Reply:
Aha czyli o to chodziło. To jeszcze jedną propozycję mam

```java
    private void replaceAvailabilitiesInOffer(Offer offerToModify, Set<Availability> availabilities) {
        Set<Availability> availabilitiesFromOffer = offerToModify.getAvailabilities();
        availabilitiesFromOffer.clear();
        availabilitiesFromOffer.addAll(availabilities);
    }
``` 

testy mi przeszły na tym więc może akurat będzie działać

Reply:
ciekawe bo nie chcialo dzialac mi, chociaz ja robilem offer.getAvai().clear()
wyprobuje jeszcze w swaggerze

Reply:
dziala xD
to ja nwm czemu wczesniej blad rzucalo xD